src/project/module.ts

```diff
@@ -4,12 +4,74 @@ import { configPath as moduleConfigPath, readConfig as readModuleConfig } from "
 import { ModuleConfig } from "../config/module.ts";
 import { Script } from "./script.ts";
 import { Actor } from "./actor.ts";
+import { Route } from "./route.ts";
 import { Project } from "./project.ts";
 import { Registry } from "./registry.ts";
 import { validateIdentifier } from "../types/identifiers/mod.ts";
 import { Casing } from "../types/identifiers/defs.ts";
-import { ProjectModuleConfig } from "../config/project.ts";
+import { configPath as projectConfigPath, ProjectModuleConfig } from "../config/project.ts";
 import { UserError } from "../error/mod.ts";
+import { RouteConfig } from "../config/module.ts";
+
+/**
+ * Validates a path provided by the user.
+ * 
+ * If the path is invalid, a {@linkcode UserError} is thrown, otherwise a version of
+ * the path without leading or trailing slashes is returned.
+ * 
+ * Paths must:
+ * - Start with a forward slash
+ * - IF they are a prefix, end with a forward slash
+ * - IF they are an exact path, NOT end with a forward slash
+ * 
+ * @param path The user-provided http path
+ * @param isPrefix Whether or not the path should be treated as a prefix
+ * @param configPath Where the (possibly) offending config file is located
+ * @returns The path without any leading or trailing slashes
+ * @throws {UserError} if the path is invalid
+ */
+function validateAndCleanPath(
```

Comment:
this should only validate the path. throw an error for anything you're cleaning here. make it so there's only 1 right way to format a path.